src/sql/workbench/services/executionPlan/common/executionPlanService.ts

```diff
@@ -27,25 +27,49 @@ export class ExecutionPlanService implements IExecutionPlanService {
 		this._providerRegisterEvent = this._onProviderRegister.event;
 	}
 
-	private async ensureCapabilitiesRegistered(): Promise<void> {
-		let providers = Object.keys(this._capabilitiesService.providers);
-		if (!providers) {
-			await new Promise<void>(resolve => {
-				this._capabilitiesService.onCapabilitiesRegistered(e => {
+	/**
+	 * This ensures that the capabilities service has registered the providers to handle execution plan requests for the given file format.
+	 * @param fileExtension Execution plan file format
+	 */
+	public async ensureFileExtensionHandlerRegistered(fileExtension: string) {
+		for (let provider in Object.keys(this._capabilitiesService.providers)) {
+			if (this._capabilitiesService.providers[provider].connection.supportedExecutionPlanFileExtensions?.includes(fileExtension)) {
+				return;
+			}
+		}
+		await (new Promise<void>((resolve, reject) => {
+			this._capabilitiesService.onCapabilitiesRegistered(e => {
```

Comment:
You should be cleaning this up by saving the disposable and then disposing it before you resolve/reject